src/NServiceBus.RabbitMQ/RabbitMQTransportInfrastructure.cs

```diff
@@ -32,7 +33,19 @@ public RabbitMQTransportInfrastructure(SettingsHolder settings, string connectio
 
             routingTopology = CreateRoutingTopology();
 
-            settings.Set(SettingsKeys.RoutingTopologySupportsDelayedDelivery, routingTopology is ISupportDelayedDelivery);
+            var routingTopologySupportsDelayedDelivery = routingTopology is ISupportDelayedDelivery;
+            settings.Set(SettingsKeys.RoutingTopologySupportsDelayedDelivery, routingTopologySupportsDelayedDelivery);
+
+            if (routingTopologySupportsDelayedDelivery)
+            {
+                var timeoutManagerFeatureDisabled = settings.GetOrDefault<FeatureState>(typeof(TimeoutManager).FullName) == FeatureState.Disabled;
+                var sendOnlyEndpoint = settings.GetOrDefault<bool>("Endpoint.SendOnly");
+
+                if (timeoutManagerFeatureDisabled || sendOnlyEndpoint)
```

Comment:
Send-only endpoints have to disable the timeout manager, because they don't start any message pumps, and the timeout manager relies on receiving messages from its satellite queues.

This is one of the prerequisites for the TimeoutManger feature to be active: https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/DelayedDelivery/TimeoutManager/TimeoutManager.cs#L22

So, if this is set, we know the timeout manage will be disabled, so it's safe to tell core we support the native delivery constraints.